### PR TITLE
Add more compatibility with upcoming 2.0 release

### PR DESCRIPTION
**Pull request by adampauls**
Builds on #34. Keeps long literals (like `0L`) and also by default sugars Numbers and Strings to their more obvious versions (just `"foo"` instead of `#(String "foo")`. Fixes some other bugs along the way.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,7 +15,6 @@
     zip_safe=False,
     install_requires=[
         "jsons==0.10.1",
-        "pydantic==1.4",
```
Comment (adampauls) [inline setup.py:18]: Fly-by: removes pydantic dataclasses in favor of the built in python ones.

File: src/dataflow/core/lispress.py [hunk 2]
```diff
@@ -175,14 +177,31 @@ def _render_lines(sexp: Lispress, max_width: int) -> List[str]:
...
         fn, *args = sexp
-        prefix = " " * NUM_INDENTATION_SPACES
-        fn_line = LEFT_PAREN + render_compact(fn)
-        arg_lines = _group_named_args(
-            [line for arg in args for line in _render_lines(arg, max_width=max_width)]
-        )
-        lines = [fn_line] + [prefix + line for line in arg_lines]
-        lines[-1] = lines[-1] + RIGHT_PAREN
-        return lines
+        if fn == VALUE_CHAR:
```
Comment (adampauls) [inline src/dataflow/core/lispress.py:180]: Special-case rendering for ^ and # syntax.

File: src/dataflow/core/lispress.py [hunk 3]
```diff
@@ -231,35 +250,42 @@ def op_to_lispress(op: Op) -> Lispress:
         value = json.loads(op.value)
         schema = value.get("schema")
         underlying = value.get("underlying")
-        # this json formatter makes it easier (than other json formatters) to tokenize the string
-        underlying_json_str = " ".join(
-            json.dumps(underlying, separators=(" ,", " : "), indent=0).split("\n")
-        )
-        return [OpType.Value.value, [schema, underlying_json_str]]
+        # Long literals look like 2L
```
Comment (adampauls) [inline src/dataflow/core/lispress.py:253]: One change: split out Long literals and also sugar up Number and Strings.

File: src/dataflow/core/lispress.py [hunk 4]
```diff
@@ -231,35 +250,42 @@ def op_to_lispress(op: Op) -> Lispress:
...
-def type_args_to_lispress(type_args: List[str]) -> Optional[Lispress]:
+def type_args_to_lispress(type_args: List[TypeName]) -> Optional[Lispress]:
     """Converts the provided list of type args into a Lispress expression."""
     if len(type_args) == 0:
         return None
     return list(map(type_name_to_lispress, type_args))
 
 
-def type_name_to_lispress(type_name: str) -> Lispress:
+def type_name_to_lispress(type_name: TypeName) -> Lispress:
```
Comment (adampauls) [inline src/dataflow/core/lispress.py:281]: Use a real class instead of a string for typenames.

File: src/dataflow/core/program_utils.py [hunk 5]
```diff
@@ -126,11 +126,11 @@ def mk_revise(
     """
     return mk_struct_op(
         schema=DataflowFn.Revise.value,
-        args={
-            ROOT_LOCATION: root_location_idx,
-            OLD_LOCATION: old_location_idx,
-            NEW: new_idx,
-        },
+        args=[
```
Comment (adampauls) [inline src/dataflow/core/program_utils.py:129]: These changes fix some mypy errors from #34.

File: src/dataflow/core/sexp.py [hunk 6]
```diff
@@ -149,4 +150,16 @@ def sexp_to_str(sexp: Sexp) -> str:
...
             return "(" + " ".join(sexp_to_str(f) for f in sexp) + ")"
     else:
-        return sexp
+        if sexp.startswith('"') and sexp.endswith('"'):
+            return sexp
+        else:
+            return _escape_symbol(sexp)
+
+
+def _escape_symbol(symbol: str) -> str:
```
Comment (adampauls) [inline src/dataflow/core/sexp.py:159]: Adds escaping for arbitrary symbols to make the parser handle the same cases as the Scala parser.

File: tests/test_dataflow/core/test_lispress.py [hunk 7]
```diff
@@ -107,7 +108,27 @@
                   (RecipientWithNameLike
                     :constraint (Constraint[Recipient])
                     :name #(PersonName "Tom"))))))))
-      :number #(Number 1))))""",
+      :number 1)))""",
+    # META_CHAR expression
```
Comment (adampauls) [inline tests/test_dataflow/core/test_lispress.py:112]: New tests.

File: tests/test_dataflow/core/test_lispress.py [hunk 8]
```diff
@@ -107,7 +108,27 @@
...
+    """
+(Yield
+  (^(Long) >
```
Comment (adampauls) [inline tests/test_dataflow/core/test_lispress.py:115]: Not the prettiest, sorry.

File: src/dataflow/core/lispress.py [hunk 9]
```diff
@@ -342,16 +368,18 @@ def _program_to_unsugared_lispress(program: Program) -> Lispress:
...
+            op_type_args_lispress = type_args_to_lispress(expression.type_args)
             op_lispress = [META_CHAR, op_type_args_lispress, op_lispress]
         curr: Sexp
         if isinstance(expression.op, (BuildStructOp, CallLikeOp)):
             curr = [op_lispress]
             named_args = get_named_args(expression)
             # if all args are named (i.e., not positional), sort them alphabetically
-            has_positional = next(filter(lambda p: p[0] is None, named_args), None) is not None
+            has_positional = (
+                next(filter(lambda p: p[0] is None, named_args), None) is not None
```
Comment (sammthomson) [inline src/dataflow/core/lispress.py:381]:
```suggestion
                any(k is None for k, _ in named_args)
```

**Review by sammthomson — APPROVED** [review]
👍

File: README-LISPRESS.md [hunk 10]
```diff
@@ -2,38 +2,64 @@
...
+(Yield 
+  (Event.id 
+    (singleton 
+      (QueryEventResponse.results 
+        (FindEventWrapperWithDefaults 
+          (Event.attendees? 
+            (AttendeeListHasRecipientConstraint 
+              (RecipientWithNameLike 
+                (^(Recipient) EmptyStructConstraint) 
+                (PersonName.apply \"janice kang\")))))))))
```
Comment (sammthomson) [inline README-LISPRESS.md:22]:
```suggestion
                (PersonName.apply "janice kang")))))))))
```

**Review by sammthomson — APPROVED** [review]
LGTM. thanks!

File: src/dataflow/core/lispress.py (hunk 9 above)
Comment (adampauls) [reply src/dataflow/core/lispress.py]: Done

**Pull request merged**